apps/console/src/features/groups/components/wizard/group-assign-users-updated.tsx

```diff
@@ -0,0 +1,331 @@
+                        selectedUserList.sort(
+                            (userObject: UserBasicInterface, comparedUserObject: UserBasicInterface) =>
+                                userObject.name?.givenName?.localeCompare(comparedUserObject.name?.givenName)
+                        );
+                    }
+                }
+            })
+            .finally(() => {
+                setIsUsersFetchRequestLoading(false);
+                onUserFetchRequestFinish();
+            });
+    };
+
+    /**
+     * The following method accepts a Map and returns the values as a string.
+     *
+     * @param attributeMap - IterableIterator<string>
+     * @returns attribute string
+     */
+    const generateAttributesString = (attributeMap: IterableIterator<string>) => {
+        const attArray: string[] = [];
+        const iterator1: IterableIterator<string> = attributeMap[ Symbol.iterator ]();
+
+        for (const attribute of iterator1) {
+            if (attribute !== "") {
+                attArray.push(attribute);
+            }
+        }
+
+        return attArray.toString();
+    };
+
+    const handleSearchFieldChange = (e: FormEvent<HTMLInputElement>, query: string, list: UserBasicInterface[],
+        stateAction: Dispatch<SetStateAction<any>>) => {
+
+        let isMatch: boolean = false;
+        const filteredRoleList: UserBasicInterface[] = [];
+
+        if (!isEmpty(query)) {
+            const regExp: RegExp = new RegExp(escapeRegExp(query), "i");
+
+            list && list.map((user: UserBasicInterface) => {
+                isMatch = regExp.test(user.userName) || (user.name && regExp.test(user.name.givenName))
+                    || (user.name && regExp.test(user.name.familyName));
+
+                if (isMatch) {
+                    filteredRoleList.push(user);
+                }
+            });
+
+            stateAction(filteredRoleList);
+
+            return;
+        }
+
+        stateAction(list);
+
+        return;
+    };
+
+    const handleAssignedItemCheckboxChange = (role: any) => {
+        const checkedGroups: any = [ ...checkedAssignedListItems ];
+
+        if (checkedGroups.includes(role)) {
+            checkedGroups.splice(checkedGroups.indexOf(role), 1);
+            setCheckedAssignedListItems(checkedGroups);
+        } else {
+            checkedGroups.push(role);
+            setCheckedAssignedListItems(checkedGroups);
+        }
+    };
+
+    const resolveListItemElement = (listItemValue: string) => {
+        return (
+            <div data-suppress="">
+                { listItemValue }
+            </div>
+        );
+    };
+
+    return (
+        <>
+            <GroupBasicsUpdated
+                data-componentid="add-group-form"
+                triggerSubmit={ triggerSubmit }
+                initialValues={ initialValues?.basic }
+                userStore={ selectedUserStore }
+                setUserStore={ setSelectedUserStore }
+                onSubmit={ (values: any) => {
```

Comment:
can't we use Map<string,FormValue> here?

Reply:
Fixed with 921cea740ad840e2bb89d86d10c54277645d24f0